protocol/app/msgs/normal_msgs.go

```diff
@@ -173,10 +177,16 @@ var (
 		"/ibc.core.client.v1.Height":                        nil,
 		"/ibc.core.client.v1.MsgCreateClient":               &ibcclient.MsgCreateClient{},
 		"/ibc.core.client.v1.MsgCreateClientResponse":       nil,
-		"/ibc.core.client.v1.MsgSubmitMisbehaviour":         &ibcclient.MsgSubmitMisbehaviour{},
+		"/ibc.core.client.v1.MsgIBCSoftwareUpgrade":         &ibcclient.MsgIBCSoftwareUpgrade{},
+		"/ibc.core.client.v1.MsgIBCSoftwareUpgradeResponse": nil,
+		"/ibc.core.client.v1.MsgRecoverClient":              &ibcclient.MsgRecoverClient{},
+		"/ibc.core.client.v1.MsgRecoverClientResponse":      nil,
```

Comment:
I never heard back so using the documentation as described above and leaving these in `normal_msgs`.

Reply:
For `MsgRecoverClient`: yes, I think we should keep this in `normal`. This is because anyone should be able to submit the gov proposal to recover a client. See [doc/comments here](https://github.com/cosmos/ibc-go/pull/4620/files#diff-e0868b09c0b7df02b3339ed6f3bedef5abd9168009812f784d6b80b94982f095R78-R93)

For `MsgIBCSoftwareUpgrade`,  yes, I also think we should keep this in `normal`. This is because it's replacing the `MsgUpgradeClient` which was the proposal submitted by users to upgrade.

Should we then move `MsgUpdateClient` and `MsgUpgradeClient` (note update vs upgrade here) to unsupported ones?

Reply:
Luke replied here: https://github.com/dydxprotocol/v4-chain/pull/867#issuecomment-1877416949